src/PhpSpec/Locator/PSR0/PSR0Locator.php

```diff
@@ -297,9 +297,10 @@ private function findSpecClassname($path)
         for ($i = 0; $i < $count; $i++) {
             if ($tokens[$i][0] === T_NAMESPACE) {
                 for ($j = $i + 1; $j < $count; $j++) {
-                    if ($tokens[$j][0] === T_STRING) {
+                    if ($tokens[$j][0] === T_STRING || $tokens[$j][0] === 314 || $tokens[$j][0] === 312) {
```

Comment:
Can't use the token names here because undefined under PHP7

Could use token_name to be more explicit, or just leave a comment?

Reply:
Please either add a comment or change this to `(\PHP_VERSION_ID >= 80000 && ($tokens[$j][0] === T_FULLY_QUALIFIED_NAME || $tokens[$j][0] === T_QUALIFIED_NAME))` (please check the constant name, I wrote them from memory).
The advantage of the second alternative is that it still uses the actual constants (and we would remove the version check in a future version of phpspec dropping support for PHP 7 by looking for version checks) while still resolving the version check at compile time to avoid any overhead (that's why I'm qualifying the constant)

Reply:
Short-circuiting the if is a great suggestion thank you